### ISSUE TITLE
🐛 Lint fix for type unUnusedImports produces a syntax error (2.0.0.-beta.6)

### ISSUE THREAD

**Issue by tajo** [biomejs/biome#6286]
### Environment information

```block
CLI:
  Version:                      2.0.0-beta.6
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v20.15.1
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/8.15.7

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Workspace:
  Open Documents:               0
```

### What happened?

```ts
import retry, { type Options } from 'async-retry';

retry();
```

The `{ type Options }` part is correctly flagged as `noUnusedImports` but

```
biome lint --fix
``` 

turns it into a syntax error:

```ts
import retryfrom 'async-retry';

retry();
```

https://next.biomejs.dev/playground/?code=aQBtAHAAbwByAHQAIAByAGUAdAByAHkALAAgAHsAIAB0AHkAcABlACAATwBwAHQAaQBvAG4AcwAgAH0AIABmAHIAbwBtACAAJwBhAHMAeQBuAGMALQByAGUAdAByAHkAJwA7AAoACgByAGUAdAByAHkAKAApADsA

### Expected result

It should output:

```ts
import retry from 'async-retry';

retry();
```

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
